### PR TITLE
Gsm descriptive errors

### PR DESCRIPTION
**Pull request by gal-forer**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://jira-hq.paloaltonetworks.local/browse/CIAC-5185

## Description
Added more descriptive errors and validations for malformed secrets.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 1]
```diff
@@ -16,17 +16,27 @@ def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest'
             logging.error(
                 f'Secret json is malformed for: {secret_id} version: {response.name.split("/")[-1]}, got error: {e}')
 
-    def list_secrets(self, project_id: str, name_filter: list = [], with_secret=False) -> list:
+    def list_secrets(self, project_id: str, name_filter: list = [], with_secret=False, attr_validation=[]) -> list:
```
Comment (dorschw) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:19]: don't use mutables (`[]`) as default args. in this case, we can use a `tuple[str, ...]` and change the square brackets in `add_secrets_file_to_build` to round ones.

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 2]
```diff
@@ -16,17 +16,27 @@ def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest'
...
                 continue
             if with_secret:
                 try:
                     secret_value = self.get_secret(project_id, secret.name)
+                    missing_attrs = []
+                    # We make sure that the keys we want in the dict are presence
+                    for attr in attr_validation:
+                        if attr not in secret_value:
+                            missing_attrs.append(attr)
+                    if len(missing_attrs) != 0:
```
Comment (dorschw) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:35]:
```suggestion
                    if missing_attrs:
```

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 3]
```diff
@@ -16,17 +16,27 @@ def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest'
...
+                        missing_attrs_str = ','.join(missing_attrs)
+                        logging.error(
+                            f'Error getting the secret: {secret.name}, it\'s missing the following required attributes: {missing_attrs_str}')  # noqa
```
Comment (dorschw) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:38]: please specify the noqa error code ignored

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 4]
```diff
@@ -16,17 +16,27 @@ def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest'
...
             if secret.name in name_filter:
                 continue
             if with_secret:
                 try:
                     secret_value = self.get_secret(project_id, secret.name)
+                    missing_attrs = []
+                    # We make sure that the keys we want in the dict are presence
+                    for attr in attr_validation:
+                        if attr not in secret_value:
+                            missing_attrs.append(attr)
```
Comment (dorschw) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:34]:
```suggestion
                    missing_attrs = [attr for attr in attr_validation if attr not in secret_value]
```

**Review by dorschw — APPROVED** [review]

File: Tests/scripts/utils/GoogleSecretManagerModule.py [hunk 5]
```diff
@@ -16,17 +16,24 @@ def get_secret(self, project_id: str, secret_id: str, version_id: str = 'latest'
...
         for secret in self.client.list_secrets(request={"parent": parent}):
             secret.name = str(secret.name).split('/')[-1]
-            logging.info(f'Getting the secret: {secret.name}')
+            logging.debug(f'Getting the secret: {secret.name}')
             if secret.name in name_filter:
                 continue
             if with_secret:
                 try:
                     secret_value = self.get_secret(project_id, secret.name)
+                    # We make sure that the keys we want in the dict are presence
```
Comment (DeanArbel) [inline Tests/scripts/utils/GoogleSecretManagerModule.py:30]:
```suggestion
                    # We make sure that the keys we want in the dict are present
```

**Review by DeanArbel — APPROVED** [review]
Very nice

**Pull request merged**